numpy/core/src/umath/ufunc_object.c

```diff
@@ -2089,6 +2107,122 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
         goto fail;
     }
 
+    if (axis) {
+        /*
+         * possibly remap axes.
+         */
+        int axis_rqst;
+        int axis_list = PyList_Check(axis);
+        int item_list = 0;
+        PyObject *item = NULL;
+
+        if (axis_list) {
+            if (PyList_Size(axis) != nop) {
+                PyErr_Format(PyExc_ValueError,
+                             "axis should be a list with %d elements", nop);
+                retval = -1;
+                goto fail;
+            }
+        } else {
+            /* a single integer is OK if there is only one core dimension */
+            if (ufunc->core_num_dim_ix > 1) {
+                PyErr_SetString(PyExc_ValueError, "axis should be a list");
+                retval = -1;
+                goto fail;
+            }
+            axis_rqst = (int)PyInt_AsSsize_t(axis);
+            if (error_converting(axis_rqst)) {
+                PyErr_Clear();
+                PyErr_SetString(PyExc_ValueError,
+                                "axis should be a list or a single integer");
+                retval = -1;
+                goto fail;
+            }
+        }
+        for (i = 0; i < nop; ++i) {
+            int op_ndim, iax, rqst, dflt;
+            int num_dims = ufunc->core_num_dims[i];
```

Comment:
Again, `num_core_dims` or `core_ndim` would be easier on my brain below, where I have to keep remembering that these are only the core dimensions...